module/zfs/zfs_ioctl.c

```diff
@@ -6822,188 +6825,215 @@ zfs_ioctl_register(const char *name, zfs_ioc_t ioc, zfs_ioc_func_t *func,
 }
 
 static void
-zfs_ioctl_register_pool(zfs_ioc_t ioc, zfs_ioc_legacy_func_t *func,
-    zfs_secpolicy_func_t *secpolicy, boolean_t log_history,
-    zfs_ioc_poolcheck_t pool_check)
+zfs_ioctl_register_pool(const char *name, zfs_ioc_t ioc,
+    zfs_ioc_legacy_func_t *func, zfs_secpolicy_func_t *secpolicy,
+    boolean_t log_history, zfs_ioc_poolcheck_t pool_check)
```

Comment:
I'd prefer not to make these zfs_ioctl changes in this PR for the sake of a slightly improved `zfs_dbgmsg()`.  I'm not against them per-say, but they obscure the issue being addressed here.  If we're interested in making the ioctl change I'd suggest moving them to their own PR and we can solicit additional feedback.